src/main/java/org/molgenis/vcf/decisiontree/filter/model/BoolClause.java

```diff
@@ -0,0 +1,25 @@
+package org.molgenis.vcf.decisiontree.filter.model;
+
+import java.util.List;
+import lombok.Builder;
+import lombok.NonNull;
+import lombok.Setter;
+import lombok.Value;
+import lombok.experimental.NonFinal;
+
+@Value
+@Builder
+public class BoolClause {
```

Comment:
The naming is confusing to me: a Query consists of clauses instead of the other way around.

Reply:
Did you consider extending BoolQuery with the AND/OR operators so that queries consisting of multiple parts can be used in a BoolNode as well?

Reply:
I agree on the confusing name, I need to figure out a suitable one though....

Reply:
I would prefer to use the new object for everything with an AND or OR, and keep the BoolNode a simple as possible.